src/target/source/codegen_source_base.cc

```diff
@@ -113,5 +113,60 @@ void CodeGenSourceBase::EndScope(int scope_id) {
   indent_ -= 2;
 }
 
+void CodeGenSourceBase::PrintType(DataType type, std::ostream& os) {  // NOLINT(*)
+  ICHECK_EQ(type.lanes(), 1) << "do not yet support vector types";
+  if (type.is_handle()) {
+    os << "void*";
+    return;
+  }
+  if (type.is_float()) {
+    if (type.bits() == 32) {
+      os << "float";
+      return;
+    }
+    if (type.bits() == 64) {
```

Comment:
should this have an `else CHECK(false)`? same question elsewhere

Reply:
I was just moving the already existing implementation to the base class :).
Moreover, there was a LOG(FATAL) at the bottom that should cover this case anyway.